src/main/java/com/exasol/adapter/dynamodb/mapping/ToDecimalPropertyToColumnValueExtractor.java

```diff
@@ -0,0 +1,83 @@
+package com.exasol.adapter.dynamodb.mapping;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import com.exasol.adapter.dynamodb.documentnode.DocumentNode;
+import com.exasol.sql.expression.BigDecimalLiteral;
+import com.exasol.sql.expression.NullLiteral;
+import com.exasol.sql.expression.ValueExpression;
+
+/**
+ * This class extracts DECIMAL values from document data like. The extraction is defined using a
+ * {@link ToDecimalPropertyToColumnMapping}.
+ */
+@java.lang.SuppressWarnings("squid:S119") // DocumentVisitorType does not fit naming conventions.
+public abstract class ToDecimalPropertyToColumnValueExtractor<DocumentVisitorType>
+        extends AbstractPropertyToColumnValueExtractor<DocumentVisitorType> {
+    private final ToDecimalPropertyToColumnMapping column;
+
+    /**
+     * Create an instance of {@link ToDecimalPropertyToColumnValueExtractor}
+     *
+     * @param column {@link ToDecimalPropertyToColumnMapping} defining the mapping
+     */
+    public ToDecimalPropertyToColumnValueExtractor(final ToDecimalPropertyToColumnMapping column) {
+        super(column);
+        this.column = column;
+    }
+
+    @Override
+    protected final ValueExpression mapValue(final DocumentNode<DocumentVisitorType> documentValue) {
+        final BigDecimal decimalValue = mapValueToDecimalWithExceptionHandling(documentValue);
+        if (decimalValue == null) {
+            return handleNotANumber();
+        } else {
+            final BigDecimal decimalWithDestinationScale = decimalValue.setScale(this.column.getDecimalScale(),
+                    RoundingMode.FLOOR);
+            if (decimalWithDestinationScale.precision() > this.column.getDecimalPrecision()) {
+                return handleOverflow();
+            } else {
+                return BigDecimalLiteral.of(decimalWithDestinationScale);
+            }
+        }
+    }
+
+    private BigDecimal mapValueToDecimalWithExceptionHandling(final DocumentNode<DocumentVisitorType> documentValue) {
+        try {
+            return mapValueToDecimal(documentValue);
+        } catch (final NumberFormatException exception) {
+            throw new ColumnValueExtractorException(
+                    "Could not read map the data source's number. Cause: " + exception.getMessage(), exception,
+                    this.column);
+        }
+    }
+
+    private ValueExpression handleNotANumber() {
+        if (this.column.getNotANumberBehaviour() == MappingErrorBehaviour.ABORT) {
+            throw new ColumnValueExtractorException(
+                    "The input value was no number. Try using a different mapping or ignore this error by setting notANumberBehaviour = \"null\".",
```

Comment:
As a user I don't understand what I should set here. Is it a property? 
And if it's a property that a user can see, I don't like the name `notANumberBehaviour`.Try to rephrase it somehow without using `A`

Reply:
How about: 
* behaviourIfNotNummeric
* NaNBehaviour
?

Reply:
I can also suggest: `notNumericBehavior`. I think `notNumberBehavior` also sounds okay, just remove that `A`